src/jit/codegenxarch.cpp

```diff
@@ -2457,7 +2457,10 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
             // X86 Long comparison
             else if (varTypeIsLong(op1Type))
             {
-                genCompareLong(treeNode);
+                if (treeNode->gtRegNum != REG_NA)
```

Comment:
Please add a comment here that for the case of not requiring to materialize result, code for compare gets generated as part of GT_JTRUE node.